src/formatDate.js

```diff
@@ -50,7 +50,44 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  function getIndex(value) {
+    return fromFormat.indexOf(value);
+  }
+
+  const newDate = [];
+  const oldDate = date.split(fromFormat[3]);
+
+  for (const item of toFormat) {
+    switch (item) {
+      case ('DD'):
+        newDate.push(oldDate[getIndex('DD')]);
```

Comment:
You make `newDate.push` each time. Maybe do it only once at the end of the function.